sdks/js/packages/core/react/components/common/upcoming-plan-change-banner/index.tsx

```diff
@@ -132,9 +136,14 @@ export function UpcomingPlanChangeBanner({
     ? getPlanIntervalName(activePlan)
     : activePlan?.title;
 
+  const showBanner =
+    nextPhase?.plan_id ||
+    (subscription?.state === SUBSCRIPTION_STATES.ACTIVE &&
+      nextPhase?.reason === 'cancel');
```

Comment:
Doubt: When is the next phase's reason `cancel`?

Reply:
when we call cancel subscription with `immediate` flag as `false`